cdap-data-fabric/src/main/java/io/cdap/cdap/data2/dataset2/lib/table/leveldb/LevelDBTableCore.java

```diff
@@ -173,10 +173,73 @@ public void persist(Map<byte[], ? extends Map<byte[], byte[]>> changes, long ver
     db.write(batch, service.getWriteOptions());
   }
 
+  /**
+   * Write the value at the target row and column with the max version {@link KeyValue.LATEST_TIMESTAMP}.
+   * as a result it hides any value written with equal or smaller version.
+   */
+  public void put(byte[] row, byte[] column, byte[] value) throws IOException {
+    getDB().put(createPutKey(row, column, KeyValue.LATEST_TIMESTAMP), value);
+  }
+
+  /**
+   * Write the value at the target row and column with the specified version.
+   */
   public void put(byte[] row, byte[] column, byte[] value, long version) throws IOException {
     getDB().put(createPutKey(row, column, version), value);
   }
 
+  /**
+   * Read the value at the target row and column with the max version {@link KeyValue.LATEST_TIMESTAMP}.
+   */
+  @Nullable
+  public byte[] get(byte[] row, byte[] column) throws IOException {
+    return getDB().get(createPutKey(row, column, KeyValue.LATEST_TIMESTAMP));
+  }
+
+  /**
+   * Read the value at the target row and column with the specified version.
+   */
+  @Nullable
+  public byte[] get(byte[] row, byte[] column, long version) throws IOException {
+    return getDB().get(createPutKey(row, column, version));
+  }
+
+  /**
+   * Read the latest (i.e. highest version) value at the target row and column.
+   */
+  @Nullable
+  public byte[] getLatest(byte[] row, byte[] col, @Nullable Transaction tx) throws IOException {
+    byte[] startKey = createStartKey(row, col);
+    byte[] endKey = createEndKey(row, upperBound(col));
+    byte[] val = null;
+    try (DBIterator iterator = getDB().iterator()) {
+      iterator.seek(startKey);
+      while (iterator.hasNext()) {
+        Map.Entry<byte[], byte[]> entry = iterator.peekNext();
+
+        // if we have reached past the endKey, nothing was found, return null
+        if (KeyValue.KEY_COMPARATOR.compare(entry.getKey(), endKey) >= 0) {
+          break;
+        }
+
+        // if this is part of a multi-row scan and we reach the next row, stop without advancing iterator
```

Comment:
Is this comment accurate? This method is only getting one value, not a multi-row scan?

Reply:
Removed the comment.